spec/acceptance/class_spec.rb

```diff
@@ -1,16 +1,61 @@
 require 'spec_helper_acceptance'
 
-describe 'nodejs class:', unless: UNSUPPORTED_PLATFORMS.include?(fact('osfamily')) do
-  it 'runs successfully' do
-    pp = <<-EOS
-    class { 'nodejs': }
-    if $::osfamily == 'RedHat' and $::operatingsystemrelease =~ /^5\.(\d+)/ {
-      class { 'epel': }
-      Class['epel'] -> Class['nodejs']
-    }
-    EOS
+describe 'nodejs class:', unless: UNSUPPORTED_PLATFORMS.include?(fact('os.family')) do
+  case fact('os.family')
+  when 'RedHat'
+    pkg_cmd = 'yum info nodejs | grep "^From repo"'
+  when 'Debian'
+    pkg_cmd = 'dpkg -s nodejs | grep ^Maintainer'
+  end
+
+  context 'default parameters' do
+    it 'runs successfully' do
+      pp = <<-EOS
+      class { 'nodejs': }
+      if $::osfamily == 'RedHat' and $::operatingsystemrelease =~ /^5\.(\d+)/ {
+        class { 'epel': }
+        Class['epel'] -> Class['nodejs']
+      }
+      EOS
+      # Run it twice and test for idempotency
+      apply_manifest(pp, catch_failures: true)
+      expect(apply_manifest(pp, catch_failures: true).exit_code).to be_zero
+    end
+
+    if fact('os.family') == 'RedHat' || fact('os.family') == 'Debian'
+      describe package('nodejs') do
+        it { is_expected.to be_installed }
+        it 'comes from the expected source' do
+          pkg_output = shell(pkg_cmd)
+          expect(pkg_output.stdout).to match 'nodesource'
+        end
+      end
+    end
+  end
+
+  context 'repo_class => epel' do
+    if fact('os.family') == 'RedHat'
```

Comment:
Can you squash this into the context: ```context 'repo_class => epel', if: fact('os.family') == 'RedHat' do``` (note that I'm not 100% sure this works so please test it)